decidim-comments/app/events/decidim/comments/comment_voted_event.rb

```diff
@@ -21,6 +26,10 @@ def downvotes
       def resource_url_params
         { anchor: "comment_#{comment.id}" }
       end
+
+      def target_resource(t_resource)
+        t_resource.is_a?(Decidim::Comments::Comment) ? target_resource(t_resource.commentable) : t_resource
```

Comment:
Could you try using `comment.root_commentable` to get to the parent and avoid the recursive call (and possible N+1s)?